hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -322,7 +326,7 @@ public ResponseCodeEnum changeOwner(NftId nftId, AccountID from, AccountID to) {
 			}
 
 			final var owner = (EntityId) nftsLedger.get(nftId, OWNER);
-			if (!owner.matches(from)) {
+			if (!owner.matches(from) && !owner.matches(AccountID.getDefaultInstance())) {
```

Comment:
When doing `Nft` transfers, the protobuf will not contain `0.0.0` for sender/receiver when the treasury transfers or receives. It will contain the treasury account, so it should not be the case that we need to do this check: `!owner.matches(AccountID.getDefaultInstance())`